components/raft_log_engine/src/engine.rs

```diff
@@ -194,6 +194,33 @@ impl FileSystem for ManagedFileSystem {
         self.base_file_system.delete(path)
     }
 
+    fn rename<P: AsRef<Path>>(&self, src_path: P, dst_path: P) -> IoResult<()> {
+        self.base_file_system
+            .rename(src_path.as_ref(), dst_path.as_ref())?;
+        if let Some(ref manager) = self.key_manager {
+            // Note: `rename` will reuse the old entryption info from `src_path`.
+            manager.link_file(
+                src_path.as_ref().to_str().unwrap(),
+                dst_path.as_ref().to_str().unwrap(),
+            )?;
```

Comment:
This is not safe. When `link_file` fails, the encryption key for `dst_path` will be lost. Refer to my implementation in https://github.com/tikv/tikv/pull/13118/commits/bc01323263766857c9d602a8b1b48b80c0882b58 (I upgraded raft-engine on 6.2 branch for a bugfix).